hrabsence/CRM/HRAbsence/Form/AbsenceRequest.php

```diff
@@ -517,320 +541,410 @@ static function formRule($fields, $files, $self) {
-      'id' => $this->_targetContactID,
-    ));
-    $mailprm[$this->_targetContactID]['display_name'] = $targetContactResult['values'][$this->_targetContactID]['display_name'];
-    $mailprm[$this->_targetContactID]['email'] = $targetContactResult['values'][$this->_targetContactID]['email'];
-
-    $domainData = civicrm_api3('Domain', 'get', ['sequential' => 1, 'current_domain' => 1]);
-    $domainEmailAddress = $domainData['values'][0]['from_email'];
+    ]);
 
-    $tplParams = array(
+    // Set template vars
+    $tplParams = [
       'messageTemplateID' => $msgTempResult['values'][$msgTempResult['id']]['id'],
-      'empName' => $mailprm[$this->_targetContactID]['display_name'],
+      'empName' => $this->_targetContact['display_name'],
       'absenceType' => $this->_absenceType,
       'absentDateDurations' => $absentDateDurations,
-      'startDate' => $submitValues['start_date'],
-      'endDate' => $submitValues['end_date'],
+      'startDate' => $this->_submitValues['start_date'],
+      'endDate' => $this->_submitValues['end_date'],
       'empPosition' => $this->_empPosition,
       'totDays' => $totDays,
       'jobHoursTime' => $this->_jobHoursTime,
-    );
+    ];
     if (!empty($appDays)) {
       $tplParams['appDays'] = $appDays;
     }
-    $sendTemplateParams = array(
+
+    // Set sendmail template params
+    $sendTemplateParams = [
       'messageTemplateID' => $tplParams['messageTemplateID'],
       'contactId' => $this->_targetContactID,
       'tplParams' => $tplParams,
-    );
+    ];
 
-    if ($this->_action & (CRM_Core_Action::ADD)) {
-      $activityParam = array(
-        'sequential' => 1,
-        'source_contact_id' => $this->_loginUserID,
-        'target_contact_id' => $this->_targetContactID,
-        'assignee_contact_id' => $this->_managerContactID,
-        'activity_type_id' => $this->_activityTypeID,
-      );
-      if (array_key_exists('_qf_AbsenceRequest_done_saveandapprove', $submitValues) || $isAdmin) {
-        $activityParam['status_id'] = CRM_Utils_Array::key('Completed', $activityStatus);
-      }
-      else {
-        //we want to keep the activity status in Scheduled for new absence if save button is clicked
-        $activityParam['status_id'] = CRM_Utils_Array::key('Scheduled', $activityStatus);
-      }
-      $result = civicrm_api3('Activity', 'create', $activityParam);
+    return $sendTemplateParams;
+  }
+  
+  /**
+   * Adds the from address to be used when sending e-mails to contacts.
+   * 
+   * @global stdClass $user
+   *   Object holding data of currently logged in user
+   * @param array $sendTemplateParams
+   *   Parameters of template to be sent as an e-mail message
+   * 
+   * @return array
+   *   Parameters of template to be sent as an e-mail message with the from
+   *   address added to it.
+   */
+  private function setFromAddress($sendTemplateParams) {
+    global $user;
+    $isAdmin = in_array('civihr_admin', $user->roles) ? true : false;
```

Comment:
since you already calculated this in postPrcoess method .. why not just passing it as a paramter instead of calculating it again ?